### ISSUE TITLE
Domain=kCFErrorDomainCFNetwork Code=310 "(null)

### ISSUE THREAD

**Issue by ZNBmm** [rs/SDWebImage#2496]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by bpoplauschi** [comment]
@ZNBmm please describe the issue you are facing. Start by filling in all the info in the issue template.

**Comment by ZNBmm** [comment]
When I used SDWebImage to load images, I found that the images couldn't be loaded. 
- This is the link to the images:
 ``https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=106923273548799&height=50&width=50&ext=1540115123&hash=AeSK9xhuO4DiFfHW``

But This link can be downloaded in Safari.

Then I used the method to see the error message
```
[[SDWebImageDownloader sharedDownloader] downloadImageWithURL:[NSURL URLWithString:@"https://platform-lookaside.fbsbx.com/platform/profilepic/?asid=106923273548799&height=50&width=50&ext=1540115123&hash=AeSK9xhuO4DiFfHW"] options:SDWebImageDownloaderAllowInvalidSSLCertificates progress:^(NSInteger receivedSize, NSInteger expectedSize, NSURL * _Nullable targetURL) {
        
    } completed:^(UIImage * _Nullable image, NSData * _Nullable data, NSError * _Nullable error, BOOL finished) {
        
    }];
```
Wrong information:
```
Error Domain=kCFErrorDomainCFNetwork Code=310 "(null)" UserInfo={_NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <68D6BE96-B0D2-4855-97A7-AE5B4170D2BF>.<1>, _kCFStreamErrorDomainKey=4, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <68D6BE96-B0D2-4855-97A7-AE5B4170D2BF>.<1>"
), _kCFStreamErrorCodeKey=-2096}
```

**Comment by dreampiggy** [comment]
See : https://developer.apple.com/documentation/cfnetwork/cfnetworkerrors/kcferrorhttpsproxyconnectionfailure

You are configurating to use HTTPS proxy. Check your proxy config firstlly, this is not anything related to SDWebImge framework.
